File: dart/dart_basico/bin/4_funcoes_metodos/funcao.dart
//calcula um array notas
double calcularNotas(List<double> notas) {
  double somarNota = 0;
  notas.forEach((nota) {
    somarNota += nota;
  });
  return somarNota / notas.length;
}

// Verifica o status da nota
String statusNota(double nota) {
  if (nota >= 7) {
    return "Aprovado";
  } else {
    return "Faça a prova final é sua ultima chance";
  }
}

//calcula nota bimestral com a prova final
double calcularProvaFinal(List<double> notas, double provaFinal) {
  double notaBimestral = calcularNotas(notas);
  return (notaBimestral + provaFinal) / 2;
}

void main() {
  const String nome = "Joab Torres Alencar";
  final List<double> notas = [7.0, 10, 3.9, 5.0];
  print("Aluno: ${nome}");
  print('Nota Geral: ${calcularNotas(notas).toStringAsFixed(2)}');
  print("Status: ${statusNota(calcularNotas(notas))}");
  print("Nota Geral + Prova Final: ${calcularProvaFinal(notas, 8).toStringAsFixed(2)}");
}

File: kotlin/PrimeiroProjeto/src/sintaxe/Funcao.kt
package sintaxe

//calcular uma lista de notas
fun calcularNotas(notas: List<Double>): Double {
    var somarnota: Double = 0.0
    notas.forEach { nota ->
        somarnota += nota
    }
    return somarnota / notas.size
}
//verifica o status da nota
fun statusNota(nota: Double): String {
    return if (nota >= 7) {
        "Aprovado"
    } else {
        "Faça a prova final e sua última chance"
    }
}
//calcula nota bimestral com a prova final
fun calcularProvaFinal(notas: List<Double>, provaFinal: Double): Double {
    val notabimestral = calcularNotas(notas)
    return (notabimestral + provaFinal) / 2
}

fun main() {
    val nome = "Joab Torres Alencar"
    val notas = listOf<Double>(7.0, 10.0, 3.9, 5.0)
    println("Aluno: ${nome}")
    println("Nota Geral: ${String.format("%.2f", calcularNotas(notas))}")
    println("Status: ${statusNota(calcularNotas(notas))}")
    println("Nota Geral + Prova Final: ${String.format("%.2f", calcularProvaFinal(notas, 8.0))}")
}
